build-system/tasks/visual-diff/index.js

```diff
@@ -444,7 +444,7 @@ async function snapshotWebpages(percy, browser, webpages) {
         let lastNavigationError;
         for (let attempt = 0; attempt < NAVIGATE_RETRIES; attempt++) {
           try {
-            return await page.goto(fullUrl, {waitUntil: 'networkidle0'});
+            return await page.goto(fullUrl, {waitUntil: 'networkidle2'});
```

Comment:
There's also a `timeout` option for `page.goto()`. Not sure if it's applicable. I tried it in #21808 (which we abandoned). WDYT?

https://github.com/rsimha/amphtml/blob/131caf58436e291ebc5b5f2fdba30842cfe3a466/build-system/tasks/visual-diff/index.js#L382-L393

Reply:
I set a default timeout for all pages here: https://github.com/rsimha/amphtml/blob/131caf58436e291ebc5b5f2fdba30842cfe3a466/build-system/tasks/visual-diff/index.js#L217
We _should_ have a finite timeout, in case the page really does pose a problem in navigation